redisson/src/main/java/org/redisson/codec/FuryCodec.java

```diff
@@ -85,22 +84,35 @@ public FuryCodec(ClassLoader classLoader, boolean registrationRequired, Language
     private final Decoder<Object> decoder = new Decoder<Object>() {
         @Override
         public Object decode(ByteBuf buf, State state) throws IOException {
-            return fury.deserialize(FuryStreamReader.of(new ByteBufInputStream(buf)));
+            MemoryBuffer furyBuffer = MemoryUtils.wrap(buf.nioBuffer());
+            try {
+                return fury.deserialize(furyBuffer);
+            } finally {
+                buf.readerIndex(buf.readerIndex() + furyBuffer.readerIndex());
+            }
         }
     };
 
     private final Encoder encoder = new Encoder() {
         @Override
         public ByteBuf encode(Object in) throws IOException {
             ByteBuf out = ByteBufAllocator.DEFAULT.buffer();
-            try {
-                ByteBufOutputStream baos = new ByteBufOutputStream(out);
-                fury.serialize(baos, in);
-                return baos.buffer();
-            } catch (Exception e) {
-                out.release();
-                throw e;
+            MemoryBuffer furyBuffer;
+            int remainingSize = out.capacity() - out.writerIndex();
+            if (out.hasArray()) {
+                furyBuffer = MemoryUtils.wrap(out.array(), out.arrayOffset() + out.writerIndex(),
+                  remainingSize);
+            } else {
+                furyBuffer =  MemoryUtils.buffer(out.memoryAddress() + out.writerIndex(), remainingSize);
```

Comment:
you need to check if memoryAddress() available with hasMemoryAddress() method

Reply:
updated